src/java.base/share/classes/java/net/SocksSocketImpl.java

```diff
@@ -297,17 +299,18 @@ public ProxySelector run() {
             URI uri;
             // Use getHostString() to avoid reverse lookups
             String host = epoint.getHostString();
-            // IPv6 literal?
             if (epoint.getAddress() instanceof Inet6Address &&
-                (!host.startsWith("[")) && (host.indexOf(':') >= 0)) {
+                    (host.indexOf(':') >= 0) &&
+                    isIPv6LiteralAddress(host)) {
```

Comment:
Is the check at line 303 still needed, now we are calling `isIPv6LiteralAddress` ?

Reply:
it's optional, as is the check for `:`. I'll remove them.